mcs/class/Mono.Compiler/Mono.Compiler.BigStep/LLVM/BitCodeEmitter.cs

```diff
@@ -0,0 +1,612 @@
+                        op, exop, operands,
+                        vm =>
+                        {
+                            LLVMValueRef tmp = LLVM.BuildSub(builder, vm.Temp0, vm.Temp1, tempName);
+                            return new NamedTempValue(tmp, tempName);
+                        });
+                    break;
+                case Opcode.Mul:
+                case Opcode.MulOvf: // TODO - Handle overflow
+                case Opcode.MulOvfUn: // TODO - Handle overflow, unsigned
+                    // tmp, tmp => tmp
+                    InvokeOperation(
+                        op, exop, operands,
+                        vm =>
+                        {
+                            LLVMValueRef tmp = LLVM.BuildMul(builder, vm.Temp0, vm.Temp1, tempName);
+                            return new NamedTempValue(tmp, tempName);
+                        });
+                    break;
+                case Opcode.Div:
+                    // tmp, tmp => tmp
+                    InvokeOperation(
+                        op, exop, operands,
+                        vm =>
+                        {
+                            LLVMValueRef tmp = LLVM.BuildFDiv(builder, vm.Temp0, vm.Temp1, tempName);
+                            return new NamedTempValue(tmp, tempName);
+                        });
+                    break;
+                case Opcode.Br:
+                case Opcode.BrS:
+                    // Special - jump to another BB
+                    LLVMBasicBlockRef bb = this.GetBranchTarget(operands[0]);
+                    LLVM.BuildBr(builder, bb);
+                    break;
+                case Opcode.DivUn:
+                    // tmp, tmp => tmp
+                    InvokeOperation(
+                        op, exop, operands,
+                        vm =>
+                        {
+                            LLVMValueRef tmp = LLVM.BuildUDiv(builder, vm.Temp0, vm.Temp1, tempName);
+                            return new NamedTempValue(tmp, tempName);
+                        });
+                    break;
+            }
+        }
+
+        /// Produce a native handle for the generated native code. Call this after Process()
+        public NativeCodeHandle Yield()
+        {
+            if (PrintDebugInfo)
+            {
+                LLVM.DumpModule(module);
+            }
+
+            if (!BitCodeEmitter.s_uninitialized)
+            {
+                lock (typeof(BitCodeEmitter))
+                {
+                    if (!BitCodeEmitter.s_uninitialized)
+                    {
+                        LLVM.LinkInMCJIT();
+                        LLVM.InitializeX86TargetMC();
+                        LLVM.InitializeX86Target();
+                        LLVM.InitializeX86TargetInfo();
+                        LLVM.InitializeX86AsmParser();
+                        LLVM.InitializeX86AsmPrinter();
+                        LLVM.InitializeMCJITCompilerOptions(s_options);
```

Comment:
this only gives you x86 (32bit), we really want amd64 (64bit) code generated. so far, the generated code happens to be simple enough that it doesn't matter.

that's how you do it:
https://github.com/lambdageek/mono/pull/23/commits/ebe23b05edbbc33de98265caa0a38402146caa90